src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetAutoMappedLocationsTable.tsx

```diff
@@ -1,61 +1,206 @@
+    setSearchTerm(term);
+    setCurrentPage(1);
+  }, 200);
+
   return (
-    <table className={styles.table} data-testid="auto-mapped">
-      <caption className="govuk-visually-hidden">
-        Table showing all auto mapped locations
-      </caption>
-      <thead>
-        <tr>
-          <th className="govuk-!-width-one-third">Current data set</th>
-          <th className="govuk-!-width-one-third">New data set</th>
-          <th>Type</th>
-          <th>Actions</th>
-        </tr>
-      </thead>
-      <tbody>
-        {/* TODO EES-5325 add search and pagination, just showing first ten for now. */}
-        {locations.slice(0, 10).map(({ candidate, mapping }) => {
-          return (
-            <tr key={`mapping-${mapping.source.code}`}>
-              <td>
-                {mapping.source.label}
-                <br />
-                <span className={styles.code}>{mapping.source.code}</span>
-              </td>
-              <td>
-                {candidate.label}
-                <br />
-                <span className={styles.code}>{candidate.code}</span>
-              </td>
-              <td>
-                <Tag colour="grey">Minor</Tag>
-              </td>
-              <td>
-                <ButtonText>
-                  Edit
-                  <VisuallyHidden>
-                    {' '}
-                    mapping for {mapping.source.label}
-                  </VisuallyHidden>
-                </ButtonText>
-              </td>
+    <>
+      <div className="govuk-grid-row">
+        <div className="govuk-grid-column-one-half">
+          <form
+            id={formId}
+            className="govuk-!-margin-bottom-2"
+            onSubmit={e => {
+              e.preventDefault();
+            }}
+          >
+            {locations.length > locationsPerPage && (
+              <FormSearchBar
+                id={`${formId}-search`}
+                label="Search auto mapped locations"
+                labelSize="s"
+                min={0}
+                name="search"
+                value={searchTerm}
+                onChange={handleSearch}
+              />
+            )}
+          </form>
+          <VisuallyHidden>
+            <p aria-atomic aria-live="polite">
+              {`${totalFilteredLocations} result${
+                totalFilteredLocations > 1 ? 's' : undefined
+              }, showing page ${currentPage} of ${totalPages}`}
+            </p>
+          </VisuallyHidden>
+        </div>
+      </div>
+      {filteredLocationChunks.length > 0 ? (
+        <table
+          className={styles.table}
+          data-testid={`auto-mapped-table-${level}`}
+        >
+          <caption className="govuk-visually-hidden">
+            Table showing all auto mapped locations
+          </caption>
+          <thead>
+            <tr>
+              <th className="govuk-!-width-one-third">Current data set</th>
+              <th className="govuk-!-width-one-third">New data set</th>
```

Comment:
These width overrides are leading to really cramped columns when the contents get long. Can we get rid of these?

![image](https://github.com/user-attachments/assets/01adce34-0207-4b0a-94a8-fc8c3857d663)

Reply:
I discussed this with Marv and we decided to keep the widths as having the text sometimes wrap onto more than two lines looked better than having tables with different column widths which can end up quite messy like:
![Screenshot 2024-08-01 151836](https://github.com/user-attachments/assets/85c9b008-cad5-4555-ae7d-9c86055e9a3a)